php-plugin/src/main/java/fr/greencodeinitiative/php/PhpRuleRepository.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2012-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package fr.greencodeinitiative.php;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableList;
+import fr.greencodeinitiative.php.checks.AvoidDoubleQuoteCheck;
+import fr.greencodeinitiative.php.checks.AvoidFullSQLRequestCheck;
+import fr.greencodeinitiative.php.checks.AvoidSQLRequestInLoopCheck;
+import fr.greencodeinitiative.php.checks.AvoidTryCatchFinallyCheck_NOK_failsAllTryStatements;
+import fr.greencodeinitiative.php.checks.AvoidUsingGlobalVariablesCheck;
+import fr.greencodeinitiative.php.checks.IncrementCheck;
+import fr.greencodeinitiative.php.checks.NoFunctionCallWhenDeclaringForLoop;
+import fr.greencodeinitiative.php.checks.UseOfMethodsForBasicOperations;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.api.server.rule.RulesDefinitionAnnotationLoader;
+import org.sonar.plugins.php.api.visitors.PHPCustomRuleRepository;
+
+public class PhpRuleRepository implements RulesDefinition, PHPCustomRuleRepository {
+
+  public static final String LANGUAGE = "php";
+  public static final String NAME = "Green Code Initiative";
+  public static final String RESOURCE_BASE_PATH = "/fr/greencodeinitiative/l10n/php/rules/custom/";
+  public static final String REPOSITORY_KEY = "gci-php";
+
+  @Override
+  public void define(Context context) {
+    NewRepository repository = context.createRepository(repositoryKey(), LANGUAGE).setName(NAME);
+
+    new RulesDefinitionAnnotationLoader().load(repository, checkClasses().toArray(new Class[] {}));
+
+    // technical debt
+    Map<String, String> remediationCosts = new HashMap<>();
+    remediationCosts.put(AvoidSQLRequestInLoopCheck.RULE_KEY, "10min");
+    remediationCosts.put(AvoidFullSQLRequestCheck.RULE_KEY, "20min");
```

Comment:
We will have to specify here if the remediation cost of a rule is different from 5 minutes? Maybe create a more modular and readable list outside the algorithm

Reply:
@utarwyn,
yes you are right.
indeed, I spend a lot of time to analyse why python and php plugin didn't work since upgrade libraries version in december 2022.
I wanted find a quick solution which turn on again these two plugins for the hackathon in order to have all plugins working for developers.
But the real solution, for me, must be found after the hackathon with more time to keep a solution like done in java plugin. For example, rollback JSON files which describe this kind of information.
I created an issue for that : #80

Reply:
Thank you for the detailed answer, I understand that we have to make the plugin work for the hackaton. My comments are only open remarks and proposals, even to be applied later. I don't have yet the hindsight to validate this code, I let Julien look at it 😉

Reply:
Hi @dedece35,

I agree with @utarwyn but as you say for the moment the important thing is to have the project to work for the hackaton. I suppose it is for the same reason you leave comments in the pom files?

The review is OK for me. I tested with the php test project and I have no problems.

Some minor things you may want to fix here (or in an other PR):

- the rule "Prefer local variables to globals" html is in French and not well formatted
- the rule "Use of methods for basic operations" has a compliant solution commented with `// NOK`

Reply:
Hi @jhertout 
thank you for feedback.
Problems corrected on rule "Prefer local variables to globals". I corrected also the exact same problem on java plugin.
On the other hand, I don't find / understand your problem on rule "Use of methods for basic operations".
See my capture : 2 lines with the problem and with "NOK" on each.
<img width="1134" alt="Capture d’écran 2023-03-23 à 22 55 32" src="https://user-images.githubusercontent.com/12011401/227376820-833d7563-2277-4706-8513-b26a9a5c7aa3.png">
If there was a problem, unit tests wouldn't pass. No ?

I merge this PR. 
We can correct your second point on other PR.